scrapy/extensions/checksettings.py

```diff
@@ -0,0 +1,54 @@
+"""
+Check settings extension
+
+See documentation in docs/topics/extensions.rst
+"""
+import logging
+import pprint
+
+from difflib import get_close_matches
+
+from scrapy import signals
+from scrapy.exceptions import NotConfigured
+
+logger = logging.getLogger(__name__)
+
+
+class CheckSettings:
+    def __init__(self, crawler):
+        self.settings = crawler.settings
+        self.not_used_settings = []
+        self.sim_min = 0.8
+
+    def get_suggestions(self):
+        setting_keys = list(self.settings.attributes.keys() - self.not_used_settings)
+        return {
+            not_used: match[0]
+            for not_used, match in [(
+                not_used,
+                get_close_matches(not_used, setting_keys, n=1, cutoff=self.sim_min))
+                for not_used in self.not_used_settings]
+            if match
+        }
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        if not crawler.settings.getbool("CHECK_SETTINGS_ENABLED"):
+            raise NotConfigured
+        ext = cls(crawler)
+        crawler.signals.connect(ext.spider_opened, signal=signals.spider_opened)
+        return ext
+
+    def spider_opened(self):
+        self.not_used_settings = [s for s in self.settings
+                                  if not self.settings.attributes[s].has_been_read]
```

Comment:
Since you are going with checking earlier, I think https://github.com/scrapy/scrapy/pull/4828#discussion_r499829754 applies even more.

In fact, it might be good to review all built-in Scrapy settings that are read at `spider_opened` or later, and include those by default into the list, maybe by providing a counterpart `_BASE` setting (see `Setting.getwithbase`).

Reply:
I added `CHECK_SETTINGS_IGNORED` setting variable. I'm not sure if a `_BASE` setting will be useful here, because if we want to check all built-in scrapy settings, the `_BASE` and the not `_BASE` setting will be the same(`[]`), please correct me if im wrong.

Reply:
Still, for the default value we should look for built-in settings that will likely be reported due to them being read after the check, if there is any.